### PR TITLE
Change `search_users` presenter

### PR DESCRIPTION
**Pull request by jsangmeister**
fixes https://github.com/OpenSlides/openslides-backend/issues/1662

waiting for https://github.com/OpenSlides/openslides-client/issues/2083

### CODE REVIEW

**Comment by jsangmeister** [comment]
Also unified the scope usages - the implementation as integers was unnecessary.

File: openslides_backend/presenter/search_users.py [hunk 1]
```diff
@@ -55,49 +54,64 @@
 )
 
 
-@register_presenter("search_users_by_name_or_email")
+@register_presenter("search_users")
 class SearchUsersByNameEmail(BasePresenter):
```
Comment (r-peschke) [inline openslides_backend/presenter/search_users.py:58]: Please rename class also

**Review by r-peschke — CHANGES_REQUESTED** [review]
Please change the class name and merge

**Review by r-peschke — APPROVED** [review]

**Comment by jsangmeister** [comment]
@r-peschke I added some more changes, mostly removing redundancies. I also made the filters hashable to circumvent the detour via tuples for the duplicate detection. Also added more tests for everything.

**Review by r-peschke — APPROVED** [review]

**Review by emanuelschuetze — CHANGES_REQUESTED** [review]
Tested and found a bug:
If an existing account has no email address (e.g. account with only first and last name: "Max Mustermann") then the account search should NOT match for "Max" and "Mustermann", because email address is not given.

Currently, the empty mail address matches by given first/last name:

```
[{"presenter":"search_users","data":{"permission_type":"meeting","permission_id":5,"search":[{"username":"","first_name":"Max","last_name":"Mustermann","email":""}]}}]
```
response:
```[[[{"id": 24, "username": "MaxMustermann", "first_name": "Max", "last_name": "Mustermann", "email": ""}]]]```

**Comment by jsangmeister** [comment]
changed to make all fields of the tuple `(first_name, last_name, email)` required.

**Review by emanuelschuetze — APPROVED** [review]
Tested

**Pull request merged**